lib/mount.c

```diff
@@ -589,7 +642,13 @@ int fuse_kern_mount(const char *mountpoint, struct mount_opts *mo)
 		goto out;
 
 	res = fuse_mount_sys(mountpoint, mo, mnt_opts);
-	if (res == -2) {
+	if (res >= 0 && mo->auto_unmount) {
+		if(0 > fuse_mount_fusermount_setup_auto_unmount_only(mountpoint, 0)) {
+			// Something went wrong, let's umount like in fuse_mount_sys.
+			umount2(mountpoint, 2); /* lazy umount */
```

Comment:
I'm not sure why we don't use `MNT_DETACH` (instead of 2) in `fuse_mount_sys`.  But I just copied the usage here.

Reply:
Will do.  I can also make a new PR to use MNT_DETACH in the other place.

Reply:
Hmm, I leave it as `2` because I am not sure the BSDs have `MNT_DETACH`.

Reply:
I think this was just introduced in mount_compat.h, so that should be fine.